src/cli/circom.rs

```diff
@@ -0,0 +1,118 @@
+use std::{fs, path::Path, process::Command};
+
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+use ansi_term::Colour::{Green, Red};
+use anyhow::{bail, Result};
+use camino::Utf8PathBuf;
+
+use crate::cli::paths::{circom_binary_path, circom_dir};
+
+const CIRCOM_VERSION: &str = "2.1.6";
+
+#[cfg(target_arch = "wasm32")]
+fn download_circom_binary(_path: impl AsRef<Path>) -> Result<Command> {
+    bail!("wasm does not support downloading")
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn download_circom_binary(path: impl AsRef<Path>) -> Result<Command> {
+    use std::io::Write;
+
+    let url = match std::env::consts::OS {
+        "linux" => "https://github.com/iden3/circom/releases/download/v2.1.6/circom-linux-amd64",
+        "macos" => "https://github.com/iden3/circom/releases/download/v2.1.6/circom-macos-amd64",
+        "windows" => {
+            "https://github.com/iden3/circom/releases/download/v2.1.6/circom-windows-amd64.exe"
+        }
+        _ => {
+            eprintln!("Unsupported OS");
+            std::process::exit(1);
```

Comment:
I would rather panic with an error message saying the Circom binary was not found *and* could not be downloaded. On an unsupported file system, it is OK for the user to provide this binary on their own.

Which makes me think this should probably modify the CircleCI config to install this binary on the Arm64 machines.

Reply:
You could even return an `Error`, as your caller would know how to handle it.